elastic/kibana
# [Maps] fix data-mapping switch enabled for vector tiles

Fixes https://github.com/elastic/kibana/issues/116297

To test
* Install sample web logs data set
* Create a new map
* Add documents layer
* Scale by vector tiles
* Style fill color by bytes.
* Click "Data mapping" button. Switch to toggle between ES data set and local data set should be disabled because MVTs from elasticsearch documents do not support getting field meta from local data.

<img width="300" alt="Screen Shot 2021-10-26 at 2 05 36 PM" src="https://user-images.githubusercontent.com/373691/138956286-90a2d107-d873-4a1d-bcf0-ada16f0cb7a3.png">

<img width="300" alt="Screen Shot 2021-10-26 at 2 27 11 PM" src="https://user-images.githubusercontent.com/373691/138956299-0aed7977-6bef-4abe-9eeb-9d4cb3b53c45.png">

This PR cleans up the IField interface with the following changes
* removed canReadFromGeoJson method. State can be obtained from source.isMvt()
* Renamed supportsFieldMeta to supportsFieldMetaFromEs to be more explicit about what this method is returning
* Added supportsFieldMetaFromLocalData. This method return true if field can get fieldMeta from local data. Fix to bug is in ESDocField's implemenation. MVT source can not get field meta from local data.
* removed supportsAutoDomain method. supportsAutoDomain is really an ORing of supportsFieldMetaFromEs and supportsFieldMetaFromLocalData. If you can get fieldMeta from either then the field supports auto domain.

PR adds supportsFieldMetaFromEs and supportsFieldMetaFromLocalData implementation for each IField class because inheritance for this type of logic has proven bug prone and following the inheritance trail made these methods more difficult to follow.
